lineapy/plugins/pipeline_writers.py

```diff
@@ -260,11 +261,24 @@ def get_task_definition(
         for var in nc.return_variables
     ]
 
-    TASK_FUNCITON_TEMPLATE = load_plugin_template("task_function.jinja")
-    return TASK_FUNCITON_TEMPLATE.render(
+    TASK_FUNCTION_TEMPLATE = load_plugin_template("task_function.jinja")
+    return TASK_FUNCTION_TEMPLATE.render(
         artifact_name=nc.safename,
         loading_blocks=input_var_loading_block,
         call_block=function_call_block,
         dumping_blocks=return_var_saving_block,
         indentation_block=" " * indentation,
     )
+
+
+class PipelineWriterFactory:
+    def get_pipeline_writer(
```

Comment:
this is a pretty basic factory method but because its wrapped inside a class, we have to initialize the class. this can probably be simplified a little by defining the get_pipeline_writer as a classmethod.

Reply:
also if you do do this, the method name can be made short (maybe simply get) because the signature will now be PipelineWriterFactory.get(Xxx)

Reply:
Updated accordingly.